### PR TITLE
Upgrade dependencies, remove unused 'rack' dependency, force UTF-8

### PR DESCRIPTION
**Pull request by ChrisBAshton**
- Can't upgrade activesupport to 6+ as Middleman requires < 6. We've upgraded to 5.2.4.3, which is the [highest pre-v6 version of activesupport](https://rubygems.org/gems/activesupport/versions).
  - Can't upgrade Middleman beyond 4.3.7 as [that's the latest version](https://rubygems.org/gems/middleman/versions), although there's a 5.0.0.rc.1 from May 2019.

Trello: https://trello.com/c/Rk03inLB/169-upgrade-dependencies-for-govuk-developer-docs

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -63,7 +63,7 @@ GEM
       faraday (~> 1.0)
     fast_blank (1.0.0)
     fastimage (2.1.7)
-    ffi (1.13.1)
+    ffi (1.12.2)
```
Comment (benthorner) [inline Gemfile.lock]: Can we apply this lock in the Gemfile, so someone in the future can avoid it?

File: Gemfile [hunk 2]
```diff
@@ -15,7 +15,7 @@ gem "middleman", "~> 4.3.7"
 gem "middleman-search_engine_sitemap", "~> 1.4"
 
 gem "github-markdown"
-gem "html-pipeline", "~> 2.5.0"
+gem "html-pipeline"
```
Comment (benthorner) [inline Gemfile:18]: Minor: could we link to the fix in the commit?

File: Gemfile.lock (hunk 1 above)
Comment (ChrisBAshton) [reply Gemfile.lock]: Good idea.

File: Gemfile (hunk 2 above)
Comment (ChrisBAshton) [reply Gemfile:19]: I did try to track down the exact version/fix but couldn't find it.

Comment (benthorner) [reply Gemfile:19]: I might be missing something here: how do we know it's fixed?

Comment (ChrisBAshton) [reply Gemfile]: Ha, your comment made me go back and run the build - everything passes. So then I switched it back to 2.11.0 (the [buggy version referenced in this issue](https://github.com/alphagov/govuk-developer-docs/issues/1821)) to confirm it was broken in that version, and it was.

So then I proceeded to increment the version and run the build, to see which version fixed the issue.
I worked my way through all of the versions and found that _none_ of them work.
Then I realised the `Gemfile.lock` was still pointing to the pinned 2.5.0 version when the build last passed 🤦 

So it does appear to still be an issue. Tomorrow I'll have a go at delving in and fixing this specific issue.

Comment (benthorner) [reply Gemfile]: Ah, I feel validated for my awkward question 😸.

Comment (ChrisBAshton) [reply Gemfile:19]: I've now found the subdependency that was causing the error, and thought we might have to [expand the context options passed to HTML::Pipeline](https://github.com/jch/html-pipeline/pull/333), but on second thoughts it's easier and more consistent to just do the UTF-8 conversion here.

**Pull request merged**